sdk/search/azure-search-documents/azure/search/documents/_internal/_search_client.py

```diff
@@ -525,11 +532,41 @@ def index_documents(self, batch, **kwargs):
         :type batch: IndexDocumentsBatch
         :rtype:  List[IndexingResult]
         """
-        index_documents = IndexBatch(actions=batch.actions)
+        return self._index_documents_actions(actions=batch.actions, **kwargs)
+
+    @distributed_trace
+    def _index_documents_actions(self, actions, **kwargs):
```

Comment:
I don't see any custom handling of timeouts. When we have multiple calls involved, we should try to avoid multiplying any timeout values passed in....